src/main/java/pl/edu/agh/iet/katabank/bankproduct/deposittype/MonthlyDepositType.java

```diff
@@ -0,0 +1,44 @@
+package pl.edu.agh.iet.katabank.bankproduct.deposittype;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.LocalDate;
+
+import static java.math.RoundingMode.HALF_DOWN;
+
+public class MonthlyDepositType implements DepositType {
+
+    private static final BigDecimal ONE_HUNDRED_PERCENT = new BigDecimal(100);
+    private static final BigDecimal TWELVE_MONTHS = new BigDecimal(12);
+    private static final int MONEY_SCALE = 2;
+    private static final int CALCULATION_SCALE = 10;
+    private static final RoundingMode ROUNDING_MODE = HALF_DOWN;
+
+    private int duration;
+    private BigDecimal yearlyInterestRatePercent;
+
+    public MonthlyDepositType(int durationInMonths, BigDecimal yearlyInterestRatePercent) {
+        this.duration = durationInMonths;
+        this.yearlyInterestRatePercent = yearlyInterestRatePercent;
```

Comment:
I think it would be worth to check for proper values of duration (not negative, not zero) and so for the yearlyInterestRate.

Reply:
(and test it)